src/opusrtp.c

```diff
@@ -163,7 +163,7 @@ ogg_packet *op_opustags(void)
     return NULL;
   }
 
-  memcpy(data, identifier, 8);
+  memcpy(data, identifier, strlen(identifier));
```

Comment:
This isn't a variable-length string (like `vendor`); it is the 64-bit signature at the beginning of the packet that identifies the kind of packet.  So it has to be 8 bytes by definition.  For consistency it would be better to change it to be like the previous function op_opushead(), removing the `identifier` variable entirely.

Reply:
Thanks for your review!

The reason why I use `strlen` to calculate the lenght of `identifier` is that `vendor`'s length is calculated by `strlen` as well.

In fact I prefer define two var `id_len` and `vendor_len` to store the length. So that it can avoid calling `strlen` for 5 times in such a simple function.

Reply:
The comment header packet must always start in the same way; at byte 0 is an 8-byte signature that identifies it as a comment header packet, at byte 8 is the vendor string length, and at byte 12 is the vendor string.  It would be inconsistent to use strlen here and then numeric literals on the following lines for the fields that should immediately follow.  However the solution is not to change them all to use strlen or a variable, because those positions are fixed; other programs that read these packets look for these fields at these specific positions, and using a variable here will not change the position at which other programs are looking for those fields.  So using strlen or a variable here would just make it easier to create broken packets that do not match what readers are looking for, and therefore cannot be read.  The size of the 64-bit signature is not stored anywhere in the packet, so it is not possible to change and it has to be a fixed size that is agreed upon by both the writer and reader; in this case the size is specified in RFC 7845.

The vendor string is different; although opusrtp currently writes a fixed string in this field, the field's length is not fixed by RFC 7845 but rather is stored in the packet.  Unlike the 64-bit signature, the length of the vendor string could be changed in opusrtp and the packet would still be readable by existing readers.  So in this case it makes sense to use strlen, even though it currently always writes the same vendor string, to make it easier to take advantage of that flexibility in the future.